lightning/src/util/sweep.rs

```diff
@@ -805,3 +811,56 @@ where
 		})
 	}
 }
+
+impl<B: Deref, F: Deref, K: Deref, L: Deref>
+	ReadableArgs<(
+		B,
+		Option<F>,
+		K,
+		L,
+		Box<
+			dyn Fn(&[&SpendableOutputDescriptor]) -> Result<Transaction, ()>
+				+ Send
+				+ Sync
+				+ 'static,
+		>,
+	)> for (BestBlock, OutputSweeper<B, F, K, L>)
+where
+	B::Target: BroadcasterInterface,
+	F::Target: Filter + Sync + Send,
+	K::Target: KVStore,
+	L::Target: Logger,
+{
+	#[inline]
+	fn read<R: io::Read>(
+		reader: &mut R,
+		args: (
+			B,
+			Option<F>,
+			K,
+			L,
+			Box<
+				dyn Fn(&[&SpendableOutputDescriptor]) -> Result<Transaction, ()>
+					+ Send
+					+ Sync
+					+ 'static,
+			>,
+		),
+	) -> Result<Self, DecodeError> {
+		let (broadcaster, chain_data_source, kv_store, logger, spend_outputs_callback) = args;
+		let state = SweeperState::read(reader)?;
+		let best_block = state.best_block;
```

Comment:
Lot of boilerplate just to pull the `best_block` out. Isn't the accessor sufficient?

Reply:
Yeah, we can hold off for now.